app/generic/connections/models.py

```diff
@@ -0,0 +1,43 @@
+from typing import Literal, Optional
+from aries_cloudcontroller import ConnRecord
+
+from pydantic import BaseModel
+
+
+class Connection(BaseModel):
+    connection_id: str
+    connection_protocol: Literal["connections/1.0", "didexchange/1.0"]
+    created_at: str
+    invitation_mode: Literal["once", "multi", "static"]
+    their_role: Literal["invitee", "requester", "inviter", "responder"]
+    state: str  # did-exchange state
+
+    my_did: Optional[str]
+    alias: Optional[str] = None
+    their_did: Optional[str] = None
+    their_label: Optional[str] = None
+    their_public_did: Optional[str] = None
+    updated_at: Optional[str] = None
+    error_msg: Optional[str] = None
+    invitation_key: Optional[str] = None
+    invitation_msg_id: Optional[str] = None
+
+
+def conn_record_to_connection(connection_record: ConnRecord):
```

Comment:
Not sure that'd improve teh code here but you could also do sth like (since you're only changing a single key - rfc23_state):

```python
connection_record = connection_record.dict()
connection_record['state'] = connection_record.pop('rfc23_state')  # changes key, keeps value
return Connection(**connection_record)
```

Reply:
Hmm, I'm in favour of creating an explicit mapping. Makes it very clear imo how we transform from the ACA-Py model to our own model.

Reply:
totally fine by me. was just a suggestion.